src/test/java/org/openrewrite/staticanalysis/EqualsAvoidsNullTest.java

```diff
@@ -64,6 +64,58 @@ public class A {
         );
     }
 
+    @DocumentExample
+    @Test
+    void invertConditional_compareTo() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              public class A {
+                  {
+                      String s = null;
+                      if(s.compareTo("test")) {}
```

Comment:
I'm wondering if we should add & then invert any comparison with `< 0` here for instance, as indicated in https://github.com/openrewrite/rewrite-static-analysis/issues/362#issuecomment-2588152102. As it stands I think we're using an `int` in an `if`. 😕

Reply:
If we decide to keep the flip for `compareTo`, we have to also migrate the comparison. Otherwise, we would also flip the semantic, see https://github.com/openrewrite/rewrite-static-analysis/issues/442.

Reply:
ATM, the implementation exclud the flip of `<` and `>` needed for `compareTo`, this can be done later.